consai_examples/consai_examples/field_observer.py

```diff
@@ -125,16 +122,12 @@ def _detection_tracked_callback(self, msg):
                 continue
             robot_id = robot.robot_id.id
             robot_is_yellow = robot.robot_id.team_color == RobotId.TEAM_COLOR_YELLOW
-            # TODO: ロボットの座標表現はリストではなくState2Dを使用する
             team_str = "our" if self._our_team_is_yellow == robot_is_yellow else "their"
             object_str = "self." + team_str + "_robots_"
-            eval(object_str + "pos")[robot_id] = [robot.pos.x, robot.pos.y]
-            eval(object_str + "angle")[robot_id] = robot.orientation
-
+            eval(object_str + "id").append(robot_id)
+            eval(object_str + "pos")[robot_id] = robot.pos
             if robot.vel:
-                eval(object_str + "vel")[robot_id] = [robot.vel[0].x, robot.vel[0].y]
-            if robot.vel_angular:
-                eval(object_str + "vel_angle")[robot_id] = robot.vel_angular
+                eval(object_str + "vel")[robot_id] = robot.vel[0]
```

Comment:
上記のコメントと同様です。vel_angularは存在するかチェックもしてください。

```suggestion
                eval(object_str + "vel")[robot_id] = State2D(x=robot.vel[0].x, y=robot.vel[0].y, theta=robot.vel_angular[0])
```

Reply:
メモ：書いた

Reply:
書いた